packages/framework/src/Component/Collector/ShopsysFrameworkDataCollector.php

```diff
@@ -111,10 +112,15 @@ public function getDocsVersion(): string
      */
     protected function resolveDocsVersion(string $versionString): string
     {
-        if (strpos($versionString, '-dev') !== false) {
-            return 'master';
+        $version = new Version($versionString);
+        if (
+            $version->hasPreReleaseSuffix()
+            && $version->getPreReleaseSuffix() === 'dev'
+            && $version->getPatch()->getValue() === 0
+            && $version->getMinor()->getValue() > 0
+        ) {
+            return sprintf('%d.%d', $version->getMajor()->getValue(), $version->getMinor()->getValue() - 1);
         }
-
-        return 'v' . $versionString;
+        return sprintf('%d.%d', $version->getMajor()->getValue(), $version->getMinor()->getValue());
```

Comment:
nitpick here.

Maybe code flow like 

```php
$version = new Version($versionString);
$versionMajorValue = $version->getMajor()->getValue();
$versionMinorValue = $version->getMinor()->getValue();

if (...) {
    $versionMinorValue--;
}

return sprintf('%d.%d', $versionMajorValue, $versionMinorValue);
```

could be more readable. But I'm ok with your solution.

Reply:
Definitely this one is better. I just shot original one as a proof of concept. 👍